### PR TITLE
add sentencepiece as a soft dependency 

### PR DESCRIPTION
**Pull request by yiyixuxu**
fix https://github.com/huggingface/diffusers/issues/9034

### CODE REVIEW

File: src/diffusers/__init__.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     is_note_seq_available,
     is_onnx_available,
     is_scipy_available,
+    is_sentencepiece_available,
```
Comment (sayakpaul) [inline src/diffusers/__init__.py:15]: Actually `sentencepiece` is needed by all the pipelines that need T5. Should we add them here as well?

File: tests/pipelines/kolors/test_kolors.py [hunk 2]
```diff
@@ -20,12 +20,11 @@
 
 from diffusers import (
     AutoencoderKL,
-    ChatGLMModel,
-    ChatGLMTokenizer,
     EulerDiscreteScheduler,
     KolorsPipeline,
     UNet2DConditionModel,
 )
+from diffusers.pipelines.kolors import ChatGLMModel, ChatGLMTokenizer
```
Comment (sayakpaul) [inline tests/pipelines/kolors/test_kolors.py:27]: I think we already install `sentencepiece` for tests. So should be good:
https://github.com/huggingface/diffusers/blob/b1f43d71897ad2c73cb891d2e92d23bc7d46a4be/setup.py#L126

**Review by sayakpaul — APPROVED** [review]
Thanks. Just a single comment.

File: tests/pipelines/kolors/test_kolors.py (hunk 2 above)
Comment (asomoza) [reply tests/pipelines/kolors/test_kolors.py:27]: yeah, that was the main reason I didn't catch this

**Review by asomoza — APPROVED** [review]
really nice! thanks.

File: src/diffusers/__init__.py (hunk 1 above)
Comment (yiyixuxu) [reply src/diffusers/__init__.py:15]: technically, they can use these pipelines without t5, no? i.e. use pre-computed text embeddings

**Pull request merged**